app/code/Magento/Variable/Model/Variable.php

```diff
@@ -162,4 +179,39 @@ public function getVariablesOptionArray($withGroup = false)
         }
         return $variables;
     }
+
+    /**
+     * Check if the given store (or the current one) has the given
+     * variable set with the given text value
+     *
+     * @param string $variableCode
+     * @param string $expectedValue
+     * @param string $type
+     * @param int $storeId
+     *
+     * @return boolean
+     */
+    public function customVariableHasValue(
```

Comment:
And then this method will be detected as unused code and simply removed...

Just FYI, not sure why PHPMD didn't work here, but any method returning `bool` should start with `is` or `has` (any method name should start with verb).